nifi-mock/src/main/java/org/apache/nifi/util/MockProcessSession.java

```diff
@@ -793,13 +798,21 @@ private void validateState(final FlowFile flowFile) {
             throw new FlowFileHandlingException(flowFile + " is not the most recent version of this flow file within this session");
         }
 
+
         for (final List<MockFlowFile> flowFiles : transferMap.values()) {
             if (flowFiles.contains(flowFile)) {
                 throw new IllegalStateException(flowFile + " has already been transferred");
             }
         }
     }
 
+    private void validateDestinationRelation(final Relationship relationship) {
+        if (!relationships.contains(relationship)) {
+            rollback();
```

Comment:
Instead of rollback what it should do is throw "IllegalArgumentException" as per API documentation found here:
  https://github.com/apache/nifi/blob/master/nifi-api/src/main/java/org/apache/nifi/processor/ProcessSession.java#L382-L383